src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -361,6 +361,14 @@ public CompilationOptions WithReportSuppressedDiagnostics(bool value)
         }
 
         /// <summary>
+        /// Creates a new options instance with the deterministic property set accordingly.
+        /// </summary>
+        public CompilationOptions WithDeterministic(bool deterministic)
```

Comment:
@TyOverby this should have triggered an API change.  Did it not? 

Also I would have expected the CSharpCompilationOptions to require "new" on the shadow of this method.